spec/integ/crypto/megolm-backup.spec.ts

```diff
@@ -134,37 +149,42 @@ describe("megolm key backups", function () {
             },
         };
 
-        return aliceTestClient!
-            .start()
-            .then(() => {
-                return createOlmSession(testOlmAccount, aliceTestClient);
-            })
-            .then(() => {
-                const privkey = decodeRecoveryKey(RECOVERY_KEY);
-                return aliceTestClient!.client!.crypto!.storeSessionBackupPrivateKey(privkey);
-            })
-            .then(() => {
-                aliceTestClient!.httpBackend.when("GET", "/sync").respond(200, syncResponse);
-                aliceTestClient!.expectKeyBackupQuery(ROOM_ID, SESSION_ID, 200, CURVE25519_KEY_BACKUP_DATA);
-                return aliceTestClient!.httpBackend.flushAllExpected();
-            })
-            .then(function (): Promise<MatrixEvent> {
-                const room = aliceTestClient!.client.getRoom(ROOM_ID)!;
-                const event = room.getLiveTimeline().getEvents()[0];
-
-                if (event.getContent()) {
-                    return Promise.resolve(event);
-                }
-
-                return new Promise((resolve, reject) => {
-                    event.once(MatrixEventEvent.Decrypted, (ev) => {
-                        logger.log(`${Date.now()} event ${event.getId()} now decrypted`);
-                        resolve(ev);
-                    });
-                });
-            })
-            .then((event) => {
-                expect(event.getContent()).toEqual("testytest");
+        fetchMock.get("express:/_matrix/client/v3/room_keys/keys/:room_id/:session_id", CURVE25519_KEY_BACKUP_DATA);
+
+        // mock for the outgoing key requests that will be sent
+        fetchMock.put("express:/_matrix/client/r0/sendToDevice/m.room_key_request/:txid", {});
+
+        fetchMock.get("express:/_matrix/client/v3/room_keys/version", CURVE25519_BACKUP_INFO);
+
+        aliceClient = await initTestClient();
+
+        // we need the backup to be trusted for the test to work
+        const backupDataToSign = JSON.parse(JSON.stringify(CURVE25519_BACKUP_INFO));
+
+        await aliceClient.crypto!.signObject(backupDataToSign.auth_data);
+        fetchMock.get("express:/_matrix/client/v3/room_keys/version", backupDataToSign, {
+            overwriteRoutes: true,
+        });
```

Comment:
if you move backupDataToSign earlier, there is no need to redo the mock:

```suggestion
        // take a copy of the backup data; we will sign it once we start the client
        const backupData = JSON.parse(JSON.stringify(CURVE25519_BACKUP_INFO));
        fetchMock.get("/_matrix/client/v3/room_keys/version", backupData);

        aliceClient = await initTestClient();

        // we need the backup to be trusted for the test to work
        await aliceClient.crypto!.signObject(backupData.auth_data);
```

Reply:
I am surprise that we can sign before calling init crypto on the client